ansible_base/jwt_consumer/common/auth.py

```diff
@@ -223,9 +227,13 @@ def process_rbac_permissions(self):
                 object_data = self.token['objects'][object_type][index]
                 resource, obj = self.get_or_create_resource(object_type, object_data)
                 if resource is not None:
-                    rd.give_permission(self.user, obj)
+                    assignment = rd.give_permission(self.user, obj)
+                    role_diff = role_diff.exclude(pk=assignment.pk)
                     logger.info(f"Granted user {self.user.username} role {object_role_name} to object {obj.name} with ansible_id {object_data['ansible_id']}")
 
+        # Remove all permissions not authorized by the JWT
+        role_diff.delete()
```

Comment:
Ways this won't work

1. The way DAB RBAC works is by filling in a "cache" table, and this logic only runs with `give_permission` and `remove_permission` calls, so this would not actually fully remove the permission. You'd need to call `remove_permission` for each case
2. For now, we support mixing and matching. Meaning, some role assignments will come from the resource provider service... but other permissions (which live in AWX/controller) are assigned directly in the AWX API. So `RoleUserAssignment.objects.filter(user=self.user)` is going to return a lot more than what you're interested to enforce... which makes me question, what _are_ we enforcing here? The JWT data only carries certain roles (5 by my last count). Since the JWT is the "authority" for those roles, you want to remove any assignments to those roles that are not in the JWT data. So this might look like `RoleUserAssignment.objects.filter(user=self.user, name__in=["Organization Member", ...])`. Exact implementation, I don't know, but that's looks like the general shape of the problem/solution.

Reply:
Thank you for the constructive feedback. I believe the changes I have made address both of the reasons this PR previously did not work. I am not sure if it is as good as it can be in terms of efficiency though.

Can you take a look through the changes again for any oversights on my end and see if there are any apparent efficiency issues?